test/Microsoft.AspNetCore.Mvc.Core.Test/Formatters/TextOutputFormatterTests.cs

```diff
@@ -233,6 +234,30 @@ public void WriteResponseContentHeaders_NoSupportedEncodings_NoEncodingIsSet()
             Assert.Equal(StatusCodes.Status406NotAcceptable, context.HttpContext.Response.StatusCode);
         }
 
+        [Fact]
+        public void GetAcceptCharsetHeaderValues_Succeeds()
+        {
+            // Arrange
+            const string testCharsetValue = "fakeValue";
+
+            var formatter = new OverrideEncodingFormatter(encoding: null);
+            var context = new DefaultHttpContext();
+            context.Request.Headers[HeaderNames.AcceptCharset] = testCharsetValue;
+
+            var writerContext = new OutputFormatterWriteContext(
+                context,
+                new TestHttpResponseStreamWriterFactory().CreateWriter,
+                objectType: null,
+                @object: null);
+
+            // Act
+            var result = TextOutputFormatter.GetAcceptCharsetHeaderValues(writerContext);
+
+            //Assert
+            Assert.Equal(1, result.Count);
+            Assert.Equal(testCharsetValue, result.Single().Value);
```

Comment:
You can simplify this either like:
```
Assert.Equals(testCharsetValue, Assert.Single(result.Count));
```

OR
```
Assert.Collection(
  result,
  value => Assert.Equals(testCharsetValue, value));
```

xUnit has opinions about how you should validate the values of collections, usually using `.Count in an assert` isn't the best option since they provide richer methods.

(I'm only mentioning this since I assume you're not that familiar with xUnit yet).

Reply:
```
var charset = Assert.Single(result);
Assert.Equal(expectedCharset, charset.Value);
```

Reply:
I would avoid doing
```csharp
Assert.Equals(testCharsetValue, Assert.Single(result.Count));
```

It's messy when one of the asserts fail as it might not be evident which one failed. Best to stick with 1 line per assert for tests IMO.

Reply:
Thanks! Done!
I am actually familiar with xUnit. I've also spent some time to update my knowledge in that area about a month ago.